### PR TITLE
Fix log path

### PR DESCRIPTION
**Pull request by Firstyear**
Fixes the log path we emit.

### CODE REVIEW

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (lslezak) [inline src/lib/authserver/ui/new_dir_inst.rb:123]: Ideally the path should be moved out from the translatable texts and shared to avoid duplication:

```ruby
def y2log
   "/var/log/YaST2/y2log"
end

Popup.Error(format(_("Failed to set up new instance! Log output may be found in %{y2log}"), y2log: y2log))


```

**Review by lslezak — COMMENTED** [review]
Note: This changes the translated texts, all translations would be invalidated. Because we are before the Leap 15.4 release and translated texts cannot be changed anymore we will merge it later (after 15.4 is released).

**Comment by dgdavid** [comment]
@Firstyear,

First of all, thank you :) 

Although the PR has been out of our radar for a couple of months, I think we can resume it and merge to master now. Are you willing to follow @lslezak's suggestions before continuing?

Regards.

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (Firstyear) [reply src/lib/authserver/ui/new_dir_inst.rb:123]: > @Firstyear,
> 
> First of all, thank you :)
> 
> Although the PR has been out of our radar for a couple of months, I think we can resume it and merge to master now. Are you willing to follow @lslezak's suggestions before continuing?
> 
> Regards.

Sure, but won't that mean I need to update every translation as well?

**Comment by Firstyear** [comment]
Sure, but won't that mean I need to update every translation as well?

**Comment by dgdavid** [comment]
> Sure, but won't that mean I need to update every translation as well?

Yes, please. I mean, every translation touched in this PR. Use the proposed solution for all of them if you don't mind.

Thanks in advance.

**Comment by Firstyear** [comment]
> > Sure, but won't that mean I need to update every translation as well?
> 
> Yes, please. I mean, every translation touched in this PR. Use the proposed solution for all of them if you don't mind.
> 
> Thanks in advance.

I already did?

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_dir_inst.rb:127]: Missing [`format`](https://ruby-doc.org/core-3.1.2/Kernel.html#method-i-format)

```suggestion
      Popup.Error(format(_('Failed to set up new instance! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_dir_inst.rb:136]: Missing [`format`](https://ruby-doc.org/core-3.1.2/Kernel.html#method-i-format)

```suggestion
        Popup.Error(format(_('Failed to set up new instance! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_dir_inst.rb:149]: Missing [`format`](https://ruby-doc.org/core-3.1.2/Kernel.html#method-i-format)

```suggestion
    Popup.Message(format(_('New instance has been set up! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:156]: Please, move path out of translation as you already did in the _new\_dir\_inst.rb_ file

```suggestion
      Popup.Error(format(_('Failed to create Kerberos KDC connection user! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_dir_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_dir_inst.rb:136]: BTW, you are always reporting the same error here. So, extracting it to a method makes more sense to me.

```ruby
def report_error
  Popup.Error(
    format(
      # TRANSLATORS: %{y2log_path} will be replaced for the YaST2 log path. Please, keep it in the translation, curly braces included.
      _('Failed to set up new instance! Log output may be found in %{y2log_path}'),
      y2log_path: '/var/log/YaST2/y2log'
    )
  )
end
```

And then just using where needed

```ruby
if !DS389.install_tls_in_nss(instance_name, tls_ca, tls_p12)
  report_error
  ...
```

Thus, you can drop the `y2log` method since you are using the right path in the proposed `report_error`. To know more about the `#TRANSLATORS` comment, please read https://yastgithubio.readthedocs.io/en/latest/localization/#notes-for-translators

But, of course, this is just a suggestion, not needed to do so for getting my approval for this PR.

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [reply src/lib/authserver/ui/new_krb_inst.rb:156]: Note: you have to create the `y2log` method here too to make it work, unless you decide to put it in some common included file.

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:163]: Please, move the path the translation.

```suggestion
      Popup.Error(format(_('Failed to create Kerberos KDC connection user! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:170]:
```suggestion
      Popup.Error(format(_('Failed to create Kerberos administration user! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:177]:
```suggestion 
      Popup.Error(format(_('Failed to create Kerberos KDC administration user! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:201]:
```suggestion
      Popup.Error(format(_('Failed to create password file! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:208]:
```suggestion
      Popup.Error(format(_('Failed to create password file! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:217]:
```suggestion
      Popup.Error(format(_('Kerberos initialisation failure! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:226]:
```suggestion
      Popup.Error(format(_('Failed to modify directory permission! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:233]:
```suggestion
      Popup.Error(format(_('Failed to modify directory permission! Log output may be found in %{y2log}'), y2log: y2log))
```

File: src/lib/authserver/ui/new_krb_inst.rb
Comment (dgdavid) [inline src/lib/authserver/ui/new_krb_inst.rb:251]:
```suggestion
    Popup.Message(format(_('New instance has been set up! Log output may be found in %{y2log}'), y2log: y2log))
```

**Review by dgdavid — COMMENTED** [review]
@Firstyear,

Sorry, I overlooked some missing [format](https://ruby-doc.org/core-3.1.2/Kernel.html#method-i-format) methods in my previous review.

I have added other suggestions too, but just moving the log path out of translations will be enough for having the PR ready to merge.

Please, let me know if you need help with those changes. I'll happily jump in to help you to move this on if needed.

**Comment by dgdavid** [comment]
> > > Sure, but won't that mean I need to update every translation as well?
> > 
> > 
> > Yes, please. I mean, every translation touched in this PR. Use the proposed solution for all of them if you don't mind.
> > Thanks in advance.
> 
> I already did?

Not in the _src/lib/authserver/ui/new\_krb\_inst.rb_ file, as far as I can see.

**Comment by dgdavid** [comment]
@Firstyear

Forgot all comments about the **src/lib/authserver/ui/new_krb_inst.rb** file. You are dropping it in https://github.com/yast/yast-auth-server/pull/74/files.

Just add those missing formats in the other file and we're done.

Thanks.

**Comment by Firstyear** [comment]
> Just add those missing formats in the other file and we're done.

In what other file? I can't find anything else that needs update ... (I'm not really a ruby programmer at all .... )

**Comment by dgdavid** [comment]
> > Just add those missing formats in the other file and we're done.
> 
> In what other file? I can't find anything else that needs update ... (I'm not really a ruby programmer at all .... )

I mean, in _src/lib/authserver/ui/new\_dir\_inst.rb_. There you followed the @lslezak suggestion, but forgetting about _format_ method. Anyway, I used the Github suggestions for helping with that. Take a look at my previous, unresolved comments. E.g., https://github.com/yast/yast-auth-server/pull/72#discussion_r930882157

**Comment by Firstyear** [comment]
> I mean, in _src/lib/authserver/ui/new_dir_inst.rb_. There you followed the @lslezak suggestion, but forgetting about _format_ method. Anyway, I used the Github suggestions for helping with that. Take a look at my previous, unresolved comments. E.g., [#72 (comment)](https://github.com/yast/yast-auth-server/pull/72#discussion_r930882157)

Ahh thanks! I didn't know about the _format method. Thanks for the help!

**Review by dgdavid — APPROVED** [review]
Since the _new_\krb\_inst.rb_ file will be removed in a subsequent PR, it LGTM. Thank you.

**Pull request merged**
